clients/filesystem-fuse/src/fuse_api_handle.rs

```diff
@@ -0,0 +1,437 @@
+        atime.map(|atime| {
+            nf.atime = atime;
+        });
+        mtime.map(|mtime| {
+            nf.mtime = mtime;
+        });
+        Ok(nf)
+    }
+}
+
+impl<T: RawFileSystem> Filesystem for FuseApiHandle<T> {
+    async fn init(&self, _req: Request) -> fuse3::Result<ReplyInit> {
+        self.local_fs.init().await;
+        Ok(ReplyInit {
+            max_write: NonZeroU32::new(16 * 1024).unwrap(),
+        })
+    }
+
+    async fn destroy(&self, _req: Request) {}
+
+    async fn lookup(
+        &self,
+        _req: Request,
+        parent: Inode,
+        name: &OsStr,
+    ) -> fuse3::Result<ReplyEntry> {
+        let file_stat = self.local_fs.lookup(parent, name.to_str().unwrap()).await?;
+        Ok(ReplyEntry {
+            ttl: self.default_ttl,
+            attr: fstat_to_file_attr(&file_stat, &self.fs_context),
+            generation: 0,
+        })
+    }
+
+    async fn getattr(
+        &self,
+        _req: Request,
+        inode: Inode,
+        fh: Option<u64>,
+        _flags: u32,
+    ) -> fuse3::Result<ReplyAttr> {
+        // check the opened file inode is the same as the inode
+        if let Some(fh) = fh {
+            self.local_fs.valid_file_id(inode, fh).await?;
+        }
+
+        let file_stat = self.local_fs.stat(inode).await?;
+        Ok(ReplyAttr {
+            ttl: self.default_ttl,
+            attr: fstat_to_file_attr(&file_stat, &self.fs_context),
+        })
+    }
+
+    async fn setattr(
+        &self,
+        req: Request,
+        inode: Inode,
+        fh: Option<u64>,
+        set_attr: SetAttr,
+    ) -> fuse3::Result<ReplyAttr> {
+        let new_file_stat = self
+            .get_modified_file_stat(inode, set_attr.size, set_attr.atime, set_attr.mtime)
+            .await?;
+        let attr = fstat_to_file_attr(&new_file_stat, &self.fs_context);
+        self.local_fs.set_attr(inode, &new_file_stat).await?;
+        Ok(ReplyAttr {
+            ttl: self.default_ttl,
+            attr: attr,
+        })
+    }
+
+    async fn mkdir(
+        &self,
+        req: Request,
+        parent: Inode,
+        name: &OsStr,
+        mode: u32,
+        umask: u32,
+    ) -> fuse3::Result<ReplyEntry> {
+        let handle_id = self
+            .local_fs
+            .create_dir(parent, name.to_str().unwrap())
+            .await?;
+        Ok(ReplyEntry {
+            ttl: self.default_ttl,
+            attr: dummy_file_attr(
+                handle_id.file_id,
+                Directory,
+                Timestamp::from(SystemTime::now()),
+                &self.fs_context,
+            ),
+            generation: 0,
+        })
+    }
+
+    async fn unlink(&self, req: Request, parent: Inode, name: &OsStr) -> fuse3::Result<()> {
+        let result = self
+            .local_fs
+            .remove_file(parent, name.to_str().unwrap())
+            .await?;
+        Ok(())
+    }
+
+    async fn rmdir(&self, req: Request, parent: Inode, name: &OsStr) -> fuse3::Result<()> {
```

Comment:
Please change the function name to `rm_dir`

Reply:
The function names are the  trait of `fues3::Filesystem`. we can't change them